### DISCUSSION TITLE
Pages not linking correctly

### DISCUSSION THREAD

**Discussion by isoebell** [community/community#147992] [New to GitHub] [answered]
### Select Topic Area

Question

### Body

Hi, I am having trouble transitioning across from VisualStudio Code to Github.

Everything in my visual studio code works seamlessly, the links are all working. However when I translate across to here for some reason the way that I have linked pages, and images etc are not working.

I thought that I had followed a similar way of website building that can be found on my github(this-instead-index.com), and everything is separated (pages, images, css etc) into folders. And even though I thought that I had mirrored it, when I enter github world my css is not applying to the pages and the images on the html files in the pages folder are not connecting.

I am unsure why my VisualStudio Code is working but it isn't translating in github, any and all feedback is welcome!

Can you see the error that I am missing? I don't want to spend hours re-organising everything if it is a small fix. What do I need to do :((((

**Comment by Lut-Lat-Aung** [comment]
Absolute paths that start with / will often look for files/folders at the root of your domain 

You can hard code like this
- Use (e.g., myusername.github.io/css/style.css) instead of your project folder (L:/project-name/css/style.css or ../css/style.css)
- Use ( myusername.github.io/image/photo.png) instead of relative paths ( ../images/photo.png)

If the site’s URL is:
https://username.github.io/my-repo-name/
add rep name

Or 

You can do this

If your css is in the same path as index.html , 
Add . . / css if it is not

link rel="stylesheet" href="./css/style.css"

You likely don’t need to rebuild or reorganize your entire project. A small path adjustment in your HTML references (and possibly fixing any case mismatches) is all. You can commit and push the changes and you are good to go.

I hope you find my answer helpful

**Reply by isoebell** [reply]
Thank you! In this instance it was the need to add ".." before the pathway, thank you!

**Comment by devafortun** [comment, marked as answer]
It seems you're experiencing issues transitioning your project from Visual Studio Code to GitHub, particularly with linked pages, images, and CSS not working as expected. Here are some potential reasons for the problem and steps to troubleshoot:
**Possible Issues**
1. File Paths:

  Relative vs Absolute Paths: Ensure that the paths to your CSS files and images are correct. When you move your project to GitHub, the structure should remain the same as in your local environment. Check if you are using relative paths correctly (e.g., ../images/picture.jpg instead of /images/picture.jpg).
  Case Sensitivity: GitHub is case-sensitive, while Windows file systems are not. Ensure that the file names and paths match exactly, including letter casing.
2. Folder Structure:
  Verify that your folder structure on GitHub matches what you have locally. If you have folders for pages, images, and CSS, they should all be in the same relative locations.
3. Deployment Method:
  If you're using GitHub Pages or another hosting service, ensure that your deployment settings are configured correctly. Sometimes, additional configurations are needed for assets to load properly.
4. Check Console for Errors:
  Open the developer console in your browser (usually F12) and check for any errors related to loading CSS or images. This can give you clues about what might be going wrong.
5. Clear Cache:
  Sometimes browsers cache old versions of files. Try clearing your browser's cache or opening the site in an incognito window to see if that resolves the issue.
6. Inspect Network Requests:
  Use the Network tab in your browser's developer tools to see if requests for CSS files and images are being made successfully or if they are returning 404 errors.
**Next Steps**
  Review Your Code: Double-check the HTML code where you link your CSS and images to ensure that everything is pointing to the correct locations.
  Test Locally: If possible, run a local server (like using live-server or similar) to see if everything works as expected before pushing changes to GitHub.
  Documentation: Refer to GitHub's documentation on [GitHub Pages](https://pages.github.com/) if you're deploying there, as it may provide additional insights into common issues.
If after trying these solutions you're still encountering problems, consider sharing specific snippets of your HTML code or file structure for more targeted assistance!

**Reply by isoebell** [reply]
Thank you! I still need to figure out why some websites I have built with a similar system worked whereas this needed the extra help, but the ".." before the pathway resolved everything! Thank you so much!
